src/socket.rs

```diff
@@ -956,6 +958,18 @@ impl Socket {
         self.inner().set_reuse_port(reuse)
     }
 
+    /// Set value for the `SO_BINDTODEVICE` option on this socket.
+    ///
+    /// If a socket is bound to an interface, only packets received from that
+    /// particular interface are processed by the socket. Note that this only
+    /// works for some socket types, particularly AF_INET sockets.
+    ///
+    /// This function is only available on Linux.
+    #[cfg(all(feature = "all", target_os = "linux"))]
+    pub fn set_bind_interface(&self, interface: CString) -> io::Result<()> {
```

Comment:
We should also have a getter, e.g. `device(&self) -> Option<CString>`. Or allow the user to pass a `CString`.